octavia_f5/controller/worker/sync_manager.py

```diff
@@ -193,23 +193,29 @@ def tenant_delete(self, network_id, device=None):
         :return: True if success, else False
         """
         tenant = m_part.get_name(network_id)
-
         if CONF.f5_agent.dry_run:
             LOG.debug("Faking tenant_delete, tenant='%s', device='%s'", tenant, device)
             return True
 
-        ret = self.bigip(device).delete(tenants=[tenant])
+        # Instead of HTTP DELETE we POST an empty tenant declaration.
+        # It works just as well, but additionally leaves us control over declaration persistency.
+        response = self.tenant_update(network_id=network_id, device=device)
+
+        # if we configure just one device, we're done
+        if device:
+            return response
 
         """ Instead of running unreliably config sync, we delete the tenant on all devices to ensure
             L2 cleanup is not blocking next config sync. """
-        if not device and CONF.f5_agent.sync_to_group:
+        passive_device = self.passive()
+        if CONF.f5_agent.sync_to_group and passive_device:
             try:
-                # Don't fail in case passive device is down, cleanup will handle this case
-                self.passive().delete(tenants=[tenant])
+                response = self.tenant_update(network_id=network_id, device=passive_device)
             except RequestException:
+                # Don't fail in case passive device is down, cleanup will handle this case
                 pass
 
-        return ret
+        return response
```

Comment:
The DELETE http call was replicated to all devices because it didn't allowed sync_to_device_group. This is not the case for the POST call which supports it. Please make sure the DELETION via POST is also replicated by device auto-sync. Then you can remove the passive-device POST completely.